components/script/dom/window.rs

```diff
@@ -1160,11 +1160,17 @@ impl WindowMethods for Window {
 
     // https://html.spec.whatwg.org/multipage/#dom-name
     fn SetName(&self, name: DOMString) {
+        if self.window_proxy.get().is_none() || self.undiscarded_window_proxy().is_none() {
+            return;
+        }
         self.window_proxy().set_name(name);
     }
 
     // https://html.spec.whatwg.org/multipage/#dom-name
     fn Name(&self) -> DOMString {
+        if self.window_proxy.get().is_none() || self.undiscarded_window_proxy().is_none() {
```

Comment:
I think we should have these changes in `windowproxy.rs`? 🤔

Reply:
I don't think so. if `self.window_proxy` is `None`, we don't have a chance to get in `windowproxy.rs`

Reply:
Since `undiscarded_window_proxy` already checks whether `window_proxy` is none, there's no need to manually check for `self.window_proxy.get()`  here.

Reply:
`undiscarded_window_proxy` assumes that `window_proxy` exists and checks whether it is discarded. However, there is a condition that `window_proxy` doesn't exist in the beginning.

Reply:
I'm not sure what you're referring to. Here's the code for `undiscarded_window_proxy`:

```rust
    pub fn undiscarded_window_proxy(&self) -> Option<DomRoot<WindowProxy>> {
        self.window_proxy.get().and_then(|window_proxy| {
            if window_proxy.is_browsing_context_discarded() {
                None
            } else {
                Some(window_proxy)
            }
        })
    }
```

It already checks for the existence of `window_proxy` and propagates the Option forward.

Reply:
yes. you are right.